Services/data/devicemodels/scripts/chiller-01-state.js

```diff
@@ -19,23 +20,35 @@ var state = {
     simulation_state: "normal_pressure"
 };
 
+// Default device properties
+var properties = {};
+
 /**
  * Restore the global state using data from the previous iteration.
  *
- * @param previousState The output of main() from the previous iteration
+ * @param previousState device state from the previous iteration
+ * @param previousProperties device properties from the previous iteration
  */
-function restoreState(previousState) {
+function restoreSimulation(previousState, previousProperties) {
     // If the previous state is null, force a default state
-    if (previousState !== undefined && previousState !== null) {
+    if (previousState) {
```

Comment:
did you verify whether this syntax is equivalent to `if (previousState !== undefined && previousState !== null) {`

Reply:
Its pretty common practice in javascript to check for both undefined and null like this. https://developer.mozilla.org/en-US/docs/Glossary/Truthy

Reply:
Yes, since undefined and null are falsey values in JS this is equivalent.

Reply:
There are differences between `null` and `truthy`:
```
var x = 0;
if (x !== undefined && x !== null) { use the value /* this will run */ }
if (x)                                               { use the value /* this will not run */ }
```

The previous code allows to use the state to store an empty string, a `false`, a `0`, a `null` etc.

The new code would discard those values and replace them with the default state.